Changed g2_prop usage - Blender 5.0

Mauii opened the pull request:
Applied changes to the G2 Properties. It is now 100% responsive.

In Blender 5.0 it was not responsive due to deprecated usage of custom id properties.

Originally:
g2_prop_name, g2_prop_tag, g2_prop_off, g2_prop_shader, g2_prop_scale

Now:
g2_prop.name, g2_prop.tag, g2_prop.off, g2_prop.shader, g2_prop.scale

Mauii committed:
Changed g2_prop usage

Applied changes to the G2 Properties usage.

In Blender 5.0 it was not responsive due to deprecated usage of custom id properties.

praviilon commented:
This is a lifesaver for Blender 5 users.

Mauii commented:
> This is a lifesaver for Blender 5 users.

I'm glad it helps you. 👌🏼😁

mrwonko committed:
Bump bl_info version to 2.0.0 for the breaking g2_prop rework below

Reworking how Ghoul 2 properties are stored on disk (to fix Blender
5.0+ compatibility, see following commits) makes files saved by the
new version unreadable by older plugin versions -- a breaking change
that warrants a major bump on its own, independent of any Blender
version support change. Documents this as a general convention in
CLAUDE.md's Releases section, alongside clarifying that storage-format
breaks (not just dropped Blender versions) warrant a major bump.

mrwonko committed:
Merge PR #77 (g2_prop PointerProperty rework) and fix its remaining bugs

Merges Mauii's Ghoul2-Properties-fix (PR #77, f5e281a), which reworks
Ghoul 2 properties from flat bpy.props-registered custom properties
(g2_prop_name, g2_prop_tag, ...) to a single g2_prop PointerProperty.
That rework is necessary: Blender 5.0 moved bpy.props-registered
properties into separate storage no longer visible via "key" in obj,
which silently broke the old flat scheme for anything created fresh in
a Blender 5.0+ session.

But the PointerProperty scheme's own existence check,
hasattr(obj, "g2_prop") and obj.g2_prop is not None, has a bug: hasattr
is always true once the type is registered, so the check only "worked"
because obj.g2_prop is not None forces a real read of the
PointerProperty -- and that read alone permanently materializes real
backing data on the object, on every Blender version. Since the G2
properties panel shows for any mesh/armature object and its draw()
touched .g2_prop unconditionally, simply selecting an unrelated object
with the Object Properties tab open silently wrote persistent data
into it, surviving even after the addon was later disabled. The
exporter's hierarchy walk and getName()'s read of ancestor objects hit
the same bug from a different angle.

Fixed by tracking configuration via a separate, genuine ad-hoc custom
property ("g2_configured", never registered through bpy.props), which
stays visible via plain dict access on every Blender version without
ever needing to touch .g2_prop merely to check it. hasG2MeshProperties/
hasG2ArmatureProperties now check that sentinel instead. The Add/Remove
operators set/clear it explicitly; import code marks it right after
legitimately writing g2_prop fields.

Also adds migration for .blend files saved under the old flat scheme:
a load_post handler reads the legacy g2_prop_* keys (still safely
readable via plain dict access even on 5.0+, since Blender's own
file-versioning duplicates old data into new storage on load) and
copies them into the new scheme, so existing rigged files keep
exporting correctly rather than silently losing all tag/name/off/
shader data.

mrwonko committed:
Add migration/no-passive-materialization/already-converted test cases

- case_migration: opens g2model.blend (old flat scheme) and asserts
  the load_post handler removed all legacy g2_prop_* keys and left at
  least one mesh and one armature configured under the new scheme.
- case_already_converted: opens a new fixture, g2model-5.0-converted.blend
  (g2model.blend re-saved after migration, so it has no legacy keys at
  all), and exports it, asserting the result matches the same
  reference GLM/GLA as case_export/case_roundtrip. Exercises the new
  scheme on its own, independent of the migration path.
- case_no_passive_materialization: the direct regression test for the
  bug this branch fixes. Creates untouched mesh/armature objects,
  checks hasG2MeshProperties/hasG2ArmatureProperties repeatedly
  (simulating panel redraws), and asserts neither custom-property keys
  nor (on 5.0+) system-storage properties change merely from checking.

Also gave g2model.blend's skeleton_root an explicit legacy
g2_prop_scale key (matching its existing 100% scale, so nothing about
the actual export changes) so case_migration has real armature
legacy data to migrate, not just mesh data -- it previously had none,
relying entirely on the PointerProperty's default.

*.blend1 (Blender's own backup files, produced incidentally while
regenerating these fixtures) added to .gitignore.

mrwonko committed:
Extend CI matrix to 4.1/5.2, update version-support docs and changelog

CI now tests the 4.1 (minimum) and 5.2 (latest) boundaries instead of
4.1/4.5 -- the fix is uniform across the whole range (confirmed via
local podman runs against 4.5/5.0/5.1 too), so exercising every
intermediate version on every CI run isn't worth the ongoing cost.

Updates CLAUDE.md's "Known Blender version support" and the manual's
installation notes to state 4.1-5.2 rather than the stale 4.1-4.5/
"5.0 known broken" claim. Adds a changelog entry under a new
`next version` placeholder heading calling out the fix and, explicitly,
that files saved by this version won't be read correctly by older
plugin versions (opening an old file still works and upgrades it
automatically).

Also adds the two pyright: ignore comments JAG2Panels.py's new
migration code needs (same reportAttributeAccessIssue pattern already
used elsewhere for g2_prop access, since the fake-bpy-module-4.5 stub
doesn't know about it).

mrwonko committed:
Merge remote-tracking branch 'origin/master' into g2-props-blender-5-compat

# Conflicts:
#	jediacademy_plugins_doc.tex

mrwonko commented on CLAUDE.md:
```diff
@@ -56,7 +56,16 @@ Versioned releases use plain SemVer (`bl_info["version"]` in `__init__.py`), dec
+
+**When a merged (or about-to-merge) change is known to require a version bump at the next release**
+(most commonly a major bump from a breaking change like the above), bump `bl_info["version"]`
+immediately in its own small PR at that time, rather than deferring it to the dedicated release-cutting
```
fold bl_info version bump commit into the main PR itself instead

mrwonko commented on tests/testdata/g2model.blend:
Why did this file change? What changed? This should not be necessary, I'd rather keep the original file.

mrwonko committed:
Address review: don't edit g2model.blend, clarify version-bump wording

- tests/testdata/g2model.blend: reverted to its original content
  (byte-identical to 8873a9e, before this branch touched it). It was
  previously edited to give skeleton_root an explicit legacy
  g2_prop_scale key, purely so case_migration had armature data to
  migrate -- but that's not this fixture's job, and the reviewer
  correctly pushed back on editing a checked-in reference file just to
  manufacture test data.
- tests/run_tests.py: case_migration now only asserts on mesh
  migration, which g2model.blend genuinely covers (it has real legacy
  mesh data). Armature-scale migration gets its own new case,
  case_migration_armature_scale, using a synthetic in-memory armature
  object with a raw legacy g2_prop_scale key instead -- same code path
  (JAG2Panels.migrateLegacyG2Props()), no fixture editing needed.
- tests/testdata/g2model-5.0-converted.blend regenerated from the
  now-reverted g2model.blend.
- CLAUDE.md: reworded the version-bump-timing convention to match what
  this branch actually did -- fold the bump into the PR introducing
  the breaking change itself, rather than presenting a separate
  dedicated PR as the primary approach.

mrwonko committed:
Drop mention of a separate version-bump PR entirely

The previous wording still described "its own small PR" as the
primary approach, with folding into the necessitating PR as a
secondary option -- reviewer wants the latter as the only guidance,
matching what this branch actually did.
